File: 1189. Maximum Number of Balloons/1189. Maximum Number of Balloons.js
/**
 * @param {string} text
 * @return {number}
 */
var maxNumberOfBalloons = function (text) {
  const letters = { b: 0, a: 0, l: 0, o: 0, n: 0 };

  for (const c of text) {
    if (c in letters) {
      letters[c]++;
    }
  }

  return Math.min(
    letters["b"],
    letters["a"],
    Math.floor(letters["l"] / 2),
    Math.floor(letters["o"] / 2),
    letters["n"]
  );
};

File: 1189. Maximum Number of Balloons.py
class Solution:
    def maxNumberOfBalloons(self, text: str) -> int:
        mapping = {"b": 0, "a": 0, "l": 0, "o": 0, "n": 0}

        for char in text:
            if char in mapping:
                mapping[char] += 1

        return min(mapping["b"], mapping["a"], mapping["l"] // 2, mapping["o"] // 2, mapping["n"])
